### ISSUE TITLE
ref prop of XAxis/YAxis incompatibility in recharts 2.13.0

### ISSUE THREAD

**Issue by binhpv** [recharts/recharts#5294]
## Do you want to request a feature or report a bug?
A bug
## What is the current behavior?
Since 2.13.0, the ref prop of XAxis and YAxis yield compile error. Look like the component is rewrite from Functional component into Class component. React class component has a ref prop that point to the instance of the class. The XAxis's Props, however, is declaring a prop from SVGElement.

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).
I have created a sandbox below. Please have a look in App.tsx line 78. There you can see that assigning the prop of type XAxisProps to XAxis is not possible.

https://codesandbox.io/p/sandbox/simple-line-chart-forked-h89tlv

## What is the expected behavior?
No compile error. The suggestion is either pass the ref from SVGElement correctly or omit it from Props or XAxis/YAxis
## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
recharts version 2.13.0 and 2.14.0 cause issue, it works with 2.12.x

**Comment by ckifer** [comment]
> Look like the component is rewrite from Functional component into Class component 

There was no re-write between 2.12.x and 2.13.0. `XAxis` has not been touched in the master branch in months https://github.com/recharts/recharts/blob/master/src/cartesian/XAxis.tsx

**Comment by binhpv** [comment]
Hi @ckifer , thanks for your quick respond.
I looked at the source code for 2.12.7, the XAxis is a functional component which mean no "ref" prop out of the box from React: https://github.com/recharts/recharts/blob/v2.12.7/src/cartesian/XAxis.tsx

In 2.14.0 it is a class component and therefore it has a "ref": https://github.com/recharts/recharts/blob/v2.14.0/src/cartesian/XAxis.tsx

**Comment by ckifer** [comment]
I see. Well it can't go back to being a function component just yet, or else we'll have a lot worse issues than this 😅 

Open to suggestions on how to keep it a class component and remove the conflict

**Comment by binhpv** [comment]
Hi @ckifer , I suggest to omit the ref at https://github.com/recharts/recharts/blob/v2.14.0/src/cartesian/XAxis.tsx#L36

The ref is not used in the code anyway.

**Comment by ckifer** [comment]
you can also just make your code 

```
type AppProps = {
  xAxisProps: React.Component<typeof XAxis>;
};
```

To match the component type

**Comment by ckifer** [comment]
Omitting the ref there would not work because the ref is not there yet. The ref is added by `React.Component` - I don't think I can omit it

**Comment by ckifer** [comment]
resolved in 2.14.1
